csv download section (#772)
dolthub/docs: content/concepts/dolthub/api.md

```

Please note: You must include a ref name (branch, tag, commit hash, etc) when making authenticated calls to the SQL API using a token. Unauthenticated API requests do not require this. They use the default branch (`main` or `master`).

## CSV API

> CSV API allow users to fetch CSV files of data from databases. We provide CSV files of database tables, individual files as well as a ZIP file of the database at a commit or branch.

Reviewer: ```suggestion
DoltHub also provides a CSV API for fetching table data as CSVs. You can request a CSV for an individual table or a zip of all table CSVs at a specified commit or branch.
```

### Example

> #### Reading

Reviewer: Don't think you need this heading since writing isn't an option

We will use an example DoltHub database, [dolthub/us-jails](https://www.dolthub.com/repositories/dolthub/us-jails/) and the Python `requests` library to explore it in the Python console.

> ##### Table

Reviewer: Maybe `One Table` here and `All Tables` instead of `Database` below

Download the table `incidents` from `main` branch:

```python
import requests
local_file = 'incidents_main.csv'
res = requests.get('https://www.dolthub.com/csv/dolthub/us-jails/main/incidents')

> with open(local_file, 'wb')as file: file.write(res.content)

Reviewer: I feel like this is easier to read for all of these if you have a line break (and space before `as`)
```suggestion
with open(local_file, 'wb') as file: 
    file.write(res.content)
```

```

Download the table `incidents` at a commit hash:

```python
import requests
local_file = 'incidents_commit.csv'
res = requests.get('https://www.dolthub.com/csv/dolthub/us-jails/u8s83gapv7ghnbmrtpm8q5es0dbl7lpd/incidents')
with open(local_file, 'wb')as file: file.write(res.content)
```

##### Database

> Download the ZIP file of the database from `main` branch:

Reviewer: ```suggestion
Download a ZIP file of all database tables from the `main` branch:
```

```python
import requests
local_file = 'us-jails_main.csv'
res = requests.get('https://www.dolthub.com/csv/dolthub/us-jails/main')
with open(local_file, 'wb')as file: file.write(res.content)
```

> Download the table `incidents` at a commit hash:

Reviewer: ```suggestion
Download a ZIP file of all database tables at a commit hash:
```

```python
import requests
local_file = 'us-jails_commit.csv'
res = requests.get('https://www.dolthub.com/csv/dolthub/us-jails/u8s83gapv7ghnbmrtpm8q5es0dbl7lpd')
with open(local_file, 'wb')as file: file.write(res.content)
```

### Authentication

API tokens can be used to authenticate calls to the CSV API over Basic Authentication. This is useful for downloading data from private databases.

You can use the token in the header when download CSVs from a private database.

```python
import requests
local_file = 'private_db_main.csv'

> res = requests.get('https://www.dolthub.com/csv/owner/private-db/main',headers={ "authorization": "token [TOKEN YOU COPIED]" },)
> with open(local_file, 'wb')as file: file.write(res.content)

Reviewer: ```suggestion
res = requests.get(
    'https://www.dolthub.com/csv/owner/private-db/main',
    headers={ "authorization": "token [TOKEN YOU COPIED]" },
)
with open(local_file, 'wb') as file: 
    file.write(res.content)
```
